lib/search/search_filters.dart

```diff
@@ -0,0 +1,71 @@
+import 'package:flutter/material.dart';
+import 'package:invidious/search/models/search_sort_by.dart';
+
+const defaultSearchSortBy = SearchSortBy.relevance;
+
+class SearchFiltersButton extends StatelessWidget {
+  final SearchSortBy initialSearchSortBy;
+  final Function(SearchSortBy) callback;
+
+  const SearchFiltersButton({
+    super.key,
+    required this.initialSearchSortBy,
+    required this.callback,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return IconButton(
+      onPressed: () {
+        showDialog(
+          context: context,
+          barrierDismissible: false,
+          builder: (BuildContext context) {
+            SearchSortBy selectedSearchSortBy = initialSearchSortBy;
+            return AlertDialog(
+              content: StatefulBuilder(
+                builder: (BuildContext context, StateSetter setState) {
+                  return SingleChildScrollView(
+                    child: ListBody(
+                      children: <Widget>[
+                        const Text('Sort by'),
+                        ...SearchSortBy.values.map((value) {
+                          return ListTile(
+                            title: Text(value.label),
+                            leading: Radio<SearchSortBy>(
+                              value: value,
+                              groupValue: selectedSearchSortBy,
+                              onChanged: (newSearchSortBy) {
+                                setState(() => selectedSearchSortBy = newSearchSortBy ?? defaultSearchSortBy);
+                              },
+                            ),
+                          );
+                        }),
+                      ]
+                    ),
+                  );
+                },
+              ),
+              actions: <Widget>[
+                TextButton(
+                  child: const Text('Cancel'),
```

Comment:
we need to use localized text here. 
Import the  translation like here: https://github.com/lamarios/clipious/blob/master/lib/home/views/screens/home.dart#L6

Then in your build function, load the locals, see:  https://github.com/lamarios/clipious/blob/master/lib/home/views/screens/home.dart#L83

To create new locals, just add it to the english one as it's the main language and let the usual heroes do the translations in other languages